File: packages/modules/api/rust/src/api/deepseek.rs
use serde::{Deserialize, Serialize};
use reqwest::Client;
use anyhow::Result;

#[derive(Serialize, Deserialize, Debug)]
pub struct ChatRequest {
    pub model: String,
    pub messages: Vec<Message>,
}

#[derive(Serialize, Deserialize, Debug)]
pub struct Message {
    pub role: String,
    pub content: String,
}

#[derive(Serialize, Deserialize, Debug)]
pub struct ChatResponse {
    pub choices: Vec<Choice>,
}

#[derive(Serialize, Deserialize, Debug)]
pub struct Choice {
    pub message: Message,
}

pub struct DeepSeekClient {
    client: Client,
    api_key: String,
}

impl DeepSeekClient {
    pub fn new(api_key: String) -> Self {
        Self { client: Client::new(), api_key }
    }

    pub async fn chat(&self, req: &ChatRequest) -> Result<ChatResponse> {
        let resp = self.client.post("https://api.deepseek.com/chat/completions")
            .header("Authorization", format!("Bearer {}", self.api_key))
            .json(req)
            .send()
            .await?
            .json::<ChatResponse>()
            .await?;
        Ok(resp)
    }
}

File: packages/modules/api/python/src/api/deepseek.py
import httpx

from .deepseek_dto import Model

BASE_URL = "https://api.deepseek.com"


class DeepSeekClient:
    def __init__(self, api_key: str) -> None:
        self._api_key = api_key
        self._client = httpx.AsyncClient()

    async def get_chat_completions(
        self, content: str, model: Model = Model.DEEPSEEK_CHAT
    ) -> dict:
        url = f"{BASE_URL}/chat/completions"
        body = {
            "model": model.value,
            "messages": [{"role": "user", "content": content}],
            "stream": False,
        }
        headers = {
            "Authorization": f"Bearer {self._api_key}",
            "Content-Type": "application/json",
        }
        response = await self._client.post(url, json=body, headers=headers)
        return response.json()
